README.md
<center> <h1>CP4WatsonAIOps CP4WAIOPS v3.7.1</h1> </center>
<center> <h2>Demo Environment Installation 🚀</h2> </center>

![K8s CNI](./doc/pics/front.png)


<center> ©2023 Niklaus Hirt / IBM </center>


<div style="page-break-after: always;"></div>


### ❗ This is provided `as-is`:

* I'm sure there are errors
* I'm sure it's not complete
* It clearly can be improved


Please contact me if you have feedback or if you find glitches or problems.

- on Slack: @niklaushirt or
- by Mail: nikh@ch.ibm.com


**❗The installation has been tested for the CP4WAIOPS v3.7.1 release on OpenShift 4.12 on:**

- OpenShift Cluster (VMware on IBM Cloud) - IPI
- OpenShift Cluster (VMware on IBM Cloud) - UPI
- IBM RedHat Openshift Kubernetes Service (ROKS) with IBM Cloud Storage (ibmc-xxx)

But it should work on other Openshift Platforms as well



> ❗Those are **non-production** installations and are suited only for demo and PoC environments. ❗
> Please refer to the official IBM Documentation for production ready installations.

<div style="page-break-after: always;"></div>

---------------------------------------------------------------
# 🚀 Quickstart
---------------------------------------------------------------


🐥 [Quick Install](#1-preparation)

* Get an OpenShift Cluster
* Get your entitlement key/pull token
* Paste the install file into the OpenShift web UI and insert your entitlement key
* Grab a coffe and come back after 2-3 hours depending on the modules you're installing

🚀 [Demo the Solution](#3-demo-the-solution)

📱 [Slack integration](#4-slack-integration) (optional)

🤓 [Demo Setup - Explained](#5-demo-setup---explained)


> ℹ️ You can find a more detailed presentation about how the automation works here: [PDF](https://ibm.box.com/s/gx0tcubl9k4phvdsrffd7taragrmvz02).
> 



Here is a quick video that walks you through the installation process
![K8s CNI](./doc/pics/JOB_INSTALL.gif)


### ⚠️ **This method creates an in-cluster installation**

- It's way faster
- You don't have to install all the tooling locally
- You don’t need a connection to the cluster during the installation (fire and forget)



> 🤓 So this could basically be done from an iPhone or iPad	

🚀 Already have a cluster? [Dive right in](#-21-install-ai-manager-with-demo-content-turbonomic-and-instana)



<div style="page-break-after: always;"></div>


---------------------------------------------------------------
# 1. Preparation
---------------------------------------------------------------


 

<div style="page-break-after: always;"></div>

<details>
<summary>✅ Prerequisites</summary>

## 1.1 Prerequisites 

### 1.1.1 OpenShift requirements 

I installed the demo in a Techzone environment.

You'll need:

- ROKS 4.12
- 5x worker nodes with **32 CPU / 128 GB**  ❗


You **might** get away with less if you don't install some components (Event Manager, ELK, Turbonomic,...) but no guarantee.



### 1.1.2 Get an OpenShift Cluster (IBMers and IBM Partners only)


IBMers and Partners can get a temporary cluster from [**Techzone**](https://techzone.ibm.com/collection/tech-zone-certified-base-images).


1. Select `OpenShift Cluster (VMware on IBM Cloud) - IPI - Public` Openshift that you want to use above

1. Create a cluster for `Practice/Self Education` or `Test` if you don't have an Opportunity Number

1. Select your preferred Geograpy

  ![K8s CNI](./doc/pics/roks01.png)

1. Select the maximum end date that fits your needs (you can extend the duration once after creation)

	![K8s CNI](./doc/pics/roks03.png)

1. Select Openshift Storage

   - Storage OCS/ODF Size: **5TiB** - This is important as with the 500MiB option your installation will run out of space rapidly.

   - OpenShift Version: **4.12**

	![K8s CNI](./doc/pics/roks02.png)

1. Select the Cluster Size

	- Worker node count: **5**
	- Flavour: **32 vCPU X 128 GB** ❗ 

	> ❗ If you want to install CP4WAIOps, Trubonomic and Instana please select **32 vCPU X 128 GB** 

	![K8s CNI](./doc/pics/roks04.png)

1. Click `Submit`
1. Once the cluster is provisioned, don't forget to extend it to 8 days if needed.


### 1.1.3 Get the CP4WAIOPS installation token (registry pull token) 

You can get the installation (registry pull token) token from [https://myibm.ibm.com/products-services/containerlibrary](https://myibm.ibm.com/products-services/containerlibrary).

This allows the CP4WAIOPS images to be pulled from the IBM Container Registry.

<div style="page-break-after: always;"></div>

</details>
<details>
<summary>⚠️ Important remarks before you start</summary>

## ⚠️⚠️ 1.2 Important remarks before you start ⚠️⚠️

Those are remarks to feedback and problem reports I got from the field.

Those scripts have been tested thoroughly on different environments and have proven to be VERY reliable.

If you think that you hit a problem:

* Make sure that you have provisioned a cluster with **5 worker nodes with 32 CPU and 128 GB** each (`b3c.16x64` - it's easy to select the wrong size). If you have Pods in `0/0` state verify the `Events`. If you get `Not enough CPU` then delete the cluster and provision the correct size.
* If you want to install CP4WAIOps, Turbonomic and Instana please select **5 worker nodes with 32 CPU and 128 GB**
* The complete installation takes about 2.5 to 8 hours depending on your region where and how you deployed ROKS to (see above).
* If you see Pods in `CrashLoop` or other error states, try to wait it out (this can be due to dependencies on other componenets that are not ready yet). Chances are that the deployment will eventually go through. If after 8h you are still stuck, ping me.
* **Select and use ONLY ONE of the scripts** below, depending on which components you want to install.




### ❗ So simply put be patient and make sure you have the correct size of cluster provisioned!

<div style="page-break-after: always;"></div>

</details>


---------------------------------------------------------------
# 2. Quick Install
---------------------------------------------------------------



You can use the scritps in the `Quick_Install` folder to rapidly spin up a demo environment.
The names should be self explaining and the headers of the files explain the modules to be installed.


You can find some examples below.


<details>
<summary>✅ Install CP4WAIOps with demo content, Turbonomic and Instana</summary>

## 2.1 Install CP4WAIOps with demo content, Turbonomic and Instana 

> ### ✅ This is probably the one that you want.

You get CP4WAIOps installed and pre-trained in one simple script.
Ready to go.
On top of that you get Turbonomic and Instana instances to play around a bit (you'll need a license key for each).



![K8s CNI](./doc/pics/install01.png)

1. In the the OCP Web UI click on the `+` sign in the right upper corner
1. Copy and paste the content from [this file](./Quick_Install/01_INSTALL_CP4WAIOPS_TURBO_INSTANA.yaml)
3. Replace `<REGISTRY_TOKEN>` at the end of the file with your pull token from step 1.1.3 (the Entitlement key from https://myibm.ibm.com)

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).


## [37.1.1] - 2023-05-04


### Major changes

- Prepare for deprecation of EventManager
- Prepare for deprecation of AIManager Name
- Complete overhaul of file structure
- Complete overhaul of naming conventions (AIManager-->CP4WAIOPS)
- OCP Console Notifications for Installation Progress


## [37.0.1] - 2023-04-14


### Major changes

- Added SockShop and ACME Air




## [37.0.1] - 2023-03-31


### Minor changes

- Probes (generic and Turbonomic)
- Ingress credentials for ITZ IPI (Thanks to Włodzimierz Dymaczewski)




## [37.0.0] - 2023-03-29

Release for 3.7.0 GA

### Major changes

- Support for CP4WAIOPS 3.7.0
- Support for Instana Datastores CRD

### Minor changes

- New RobotShop Overlay Topology





## [36.0.1] - 2022-01-20

### Minor changes

- WebHook Probe Added



## [36.0.1] - 2022-01-16

### Fixes

- A lot of small bugfixes
### Minor changes

- Added EDA (Event Driven Ansible) Demo scripts for Turbo
- Added custom, non-official Discord integrations 





## [36.0.2] - 2022-12-21

Changes for Cloud Pak Deployer compatibility




## [36.0.1] - 2022-12-20

### Fixes

- Fix for Event Manager Version
### Minor changes

- Added encrypted Instana and Trubonomic licenses





## [36.0.0] - 2022-12-14

Release for 3.6.0 GA

### Major changes

- Support for CP4WAIOPS 3.6.0

### Minor changes

- Added Business Criticalities
- Option to freeze the Catalog version





## [35.1.6] - 2022-12-06

### Fixes

- Fixed patching for valid ingress certificates




## [35.1.5] - 2022-11-17

### Major changes

- Added CP4WAIOps Cards
- Added Turbonomic Demo Content




## [35.1.4] - 2022-11-15

### Major changes

- Added Instana to the mix (mostly stolen from Luca Floris - thanks!!!)




## [35.1.3] - 2022-11-11

### Major changes

- A lot of consolidation for integration into https://github.com/IBM/cloud-pak-deployer



## [35.1.2] - 2022-10-27

### Major changes

- Corrected a bug with links in the Demo UI 
- Secured routes for RobotShop, LDAP, Spark



## [35.1.1] - 2022-10-26

### Major changes

- Compatible with 3.5.1
- Added possibility to use custom sizing configurations



## [35.0.5] - 2022-10-23

### Major changes

- Update to DemoUI - new Interface design to match Techzone



## [35.0.5] - 2022-10-21

### Major changes

- Update to DemoUI - separate Namespace
- Corrected blocking installation for DemoUI



## [35.0.4] - 2022-10-12

### Major changes

- OpenLDAP with persistence - thanks Włodzimierz Dymaczewski



## [35.0.3] - 2022-09-30

### Major changes

- Added scripts to check installation



## [35.0.2] - 2022-09-28

Release for 3.5.0 GA

### Major changes

- Support for CP4WAIOPS 3.5.0




## [35.0.1] - 2022-09-08

Realease for 3.5.0 FVT on OCP VPC 2

### Major changes

- Support for OCS/ODF
- New AWX Version 
- Corrected AWX execution bug




## [35.0.0] - 2022-08-29

Realease for 3.5.0 FVT 

### Major changes

- Support for disruptionCostPerMin
- Adapted LAD model






## [34.1.0] - 2022-06-29

Realease for 3.4.1 GA. 

### Major changes

- Support for version WAIOPS 3.4.1
- Re-added fixed ChangeRisk training






## [34.0.2] - 2022-06-29

Realease for 3.4.0 GA. 
Some specifics apply: please see the README file.

### Major changes

- Support for version WAIOPS 3.4
- Completely revamped the Ansible project structure


tools/02_awx-init.sh
#!/bin/bash
echo "*****************************************************************************************************************************"
echo " 🐥 CloudPak for Watson AIOPs - Configure AWX"
echo "*****************************************************************************************************************************"
echo "  "

export AWX_ROUTE=$(oc get route -n awx awx -o jsonpath={.spec.host})
export AWX_URL=$(echo "https://$AWX_ROUTE")
echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🕦  Wait for AWX being ready"
while : ; do
      READY=$(curl -s $AWX_URL|grep "Application is not available")
      if [[  $READY  =~ "Application is not available" ]]; then
            echo "        AWX not ready yet. Waiting 15 seconds"
            sleep 30
      else
            break
      fi
done
echo "       ✅  OK: AWX ready"


echo ""
echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🛠️   Initialisation"

export AWX_ROUTE=$(oc get route -n awx awx -o jsonpath={.spec.host})
export ADMIN_USER=admin
export ADMIN_PASSWORD=$(oc -n awx get secret awx-admin-password -o jsonpath='{.data.password}' | base64 --decode && echo)

export OCP_URL=https://c108-e.eu-gb.containers.cloud.ibm.com:30553
export OCP_TOKEN=CHANGE-ME

export AWX_REPO=$INSTALL_REPO
export RUNNER_IMAGE=quay.io/niklaushirt/cp4waiops-awx:0.1.4


echo "       ✅  Done"


echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🔎  Parameters"
echo "        🧔 ADMIN_USER:                $ADMIN_USER"
echo "        🔐 ADMIN_PASSWORD:            $ADMIN_PASSWORD"
echo "        🌏 AWX_ROUTE:                 $AWX_ROUTE"
echo ""     
echo "        📥 SHOW_TOOLS:                $SHOW_TOOLS"
echo "        📥 SHOW_ADDONS:               $SHOW_ADDONS"
echo "        📥 SHOW_CONFIG:               $SHOW_CONFIG"
echo "        📥 SHOW_DEBUG:                $SHOW_DEBUG"
echo ""
echo ""
echo ""

echo "        🔐 cp_entitlement_key:     $cp_entitlement_key"


echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create AWX Execution Environment"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/execution_environments/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "CP4WAIOPS Execution Environment",
    "description": "CP4WAIOPS Execution Environment",
    "organization": null,
    "image": "'$RUNNER_IMAGE'",
    "credential": null,
    "pull": "missing"
}')

if [[ $result =~ " already exists" ]];
then
    export EXENV_ID=$(curl -X "GET" -s "https://$AWX_ROUTE/api/v2/execution_environments/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure|jq -c '.results[]| select( .name == "CP4WAIOPS Execution Environment")|.id')
    echo "        Already exists with ID:$EXENV_ID"
else
    echo "        Execution Environment created: "$(echo $result|jq ".created")
    export EXENV_ID=$(echo $result|jq ".id")
fi 


echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create AWX Project"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/projects/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "CP4WAIOPS ANSIBLE INSTALLER",
    "description": "CP4WAIOPS ANSIBLE INSTALLER",
    "local_path": "",
    "scm_type": "git",
    "scm_url": "'$AWX_REPO'",
    "scm_branch": "",
    "scm_refspec": "",
    "scm_clean": false,
    "scm_track_submodules": false,
    "scm_delete_on_update": false,
    "credential": null,
    "timeout": 0,
    "organization": 1,
    "scm_update_on_launch": false,
    "scm_update_cache_timeout": 0,
    "allow_override": false,
    "default_environment": null
}')

if [[ $result =~ " already exists" ]];
then
    export PROJECT_ID=$(curl -X "GET" -s "https://$AWX_ROUTE/api/v2/projects/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure|jq -c '.results[]| select( .name == "CP4WAIOPS ANSIBLE INSTALLER")|.id')
    echo "        Already exists with ID:$PROJECT_ID"
else
    echo "        Project created: "$(echo $result|jq ".created")
    export PROJECT_ID=$(echo $result|jq ".id")
fi



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create AWX Inventory"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/inventories/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "CP4WAIOPS Install",
    "description": "CP4WAIOPS Install",
    "organization": 1,
    "project": '$PROJECT_ID',
    "kind": "",
    "host_filter": null,
    "variables": "---\nOCP_LOGIN: false\nOCP_URL: '$OCP_URL'\nOCP_TOKEN: '$OCP_TOKEN'\n#cp_entitlement_key: '$cp_entitlement_key'"
}
')

if [[ $result =~ " already exists" ]];
then
    export INVENTORY_ID=$(curl -X "GET" -s "https://$AWX_ROUTE/api/v2/inventories/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure|jq -c '.results[]| select( .name == "CP4WAIOPS Install")|.id')
    echo "        Already exists with ID:$INVENTORY_ID"
else
    echo "        Inventory created: "$(echo $result|tr -d '\n'|jq ".created")
    export INVENTORY_ID=$(echo $result|tr -d '\n'|jq ".id")
    echo ""
    echo "   ------------------------------------------------------------------------------------------------------------------------------"
    echo "   🕦  Waiting 15s"
    sleep 15
fi


echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   ✅  RPOJECT Parameters"
echo "        🧔 EXECUTION_ENV:             $EXENV_ID"
echo "        🔐 INVENTORY_ID:              $INVENTORY_ID"
echo "        🌏 PROJECT_ID:                $PROJECT_ID"
echo ""






echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS AI Manager And Demo Content - 3.3.2"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "01_Install CP4WAIOPS AI Manager And Demo Content - 3.3.2",
    "description": "Install CP4WAIOPS AI Manager And Demo Content - 3.3.2",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-aimanager-all-33.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 


echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS AI Manager Only - 3.3.2"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "02_Install CP4WAIOPS AI Manager Only - 3.3.2",
    "description": "Install CP4WAIOPS AI Manager Only - 3.3.2",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-aimanager-33.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 




echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS AI Manager Demo Content - 3.3.2"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "03_Install CP4WAIOPS AI Manager Demo Content - 3.3.2",
    "description": "Install CP4WAIOPS AI Manager Demo Content - 3.3.2",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-aimanager-demo-content-33.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS Event Manager - 3.3.2"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "04_Install CP4WAIOPS Event Manager - 3.3.2",
    "description": "02_Install CP4WAIOPS Event Manager - 3.3.2",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-eventmanager-33.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS Infrastructure Management - 3.3.2"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "05_Install CP4WAIOPS Infrastructure Management - 3.3.2",
    "description": "03_Install CP4WAIOPS Infrastructure Management - 3.3.2",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-inframgt-33.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')

if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 




echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS AI Manager And Demo Content - 3.4 FVT"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "01_Install CP4WAIOPS AI Manager And Demo Content - 3.4 FVT",
    "description": "Install CP4WAIOPS AI Manager And Demo Content - 3.4 FVT",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-aimanager-all-34.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 


echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS AI Manager Only - 3.4 FVT"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "02_Install CP4WAIOPS AI Manager Only - 3.4 FVT",
    "description": "Install CP4WAIOPS AI Manager Only - 3.4 FVT",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-aimanager-34.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 




echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS AI Manager Demo Content - 3.4 FVT"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "03_Install CP4WAIOPS AI Manager Demo Content - 3.4 FVT",
    "description": "Install CP4WAIOPS AI Manager Demo Content - 3.4 FVT",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-aimanager-demo-content-34.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS Event Manager - 3.4 FVT"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "04_Install CP4WAIOPS Event Manager - 3.4 FVT",
    "description": "02_Install CP4WAIOPS Event Manager - 3.4 FVT",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-eventmanager-34.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')
if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install CP4WAIOPS Infrastructure Management - 3.4 FVT"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "05_Install CP4WAIOPS Infrastructure Management - 3.4 FVT",
    "description": "03_Install CP4WAIOPS Infrastructure Management - 3.4 FVT",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID',
    "ask_variables_on_launch": true,
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-inframgt-34.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'"
}
')

if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Get CP4WAIOPS Logins"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "10_Get CP4WAIOPS Logins",
    "description": "10_Get CP4WAIOPS Logins",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/90_get-all-logins.yaml",
    "scm_branch": "",
    "extra_vars": "",
    "execution_environment": '$EXENV_ID'
}
')

if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install Turbonomic"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "10_Install Turbonomic",
    "description": "Install Turbonomic",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-turbonomic.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'",
    "execution_environment": '$EXENV_ID'
}
')

if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo ""
echo "   ------------------------------------------------------------------------------------------------------------------------------"
echo "   🚀  Create Job: Install ELK"
export result=$(curl -X "POST" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure \
-H 'content-type: application/json' \
-d $'{
    "name": "11_Install ELK",
    "description": "Install ELK",
    "job_type": "run",
    "inventory": '$INVENTORY_ID',
    "project": '$PROJECT_ID',
    "playbook": "ansible/00_cp4waiops-install.yaml",
    "scm_branch": "",
    "extra_vars": "---\nconfig_file_path: \"./configs/cp4waiops-roks-elk.yaml\"\ncp_entitlement_key: '$cp_entitlement_key'",
    "execution_environment": '$EXENV_ID'
}
')

if [[ $result =~ " already exists" ]];
then
    echo "        Already exists."
else
    echo "        Job created: "$(echo $result|jq ".created")
fi 



echo "    "
echo "    "
echo "    "
echo "    "
echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo " 🔎 AWX Installed Content"
echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo "    "
echo "    -----------------------------------------------------------------------------------------------------------------------------------------------"
echo "    📥 AWX Projects"
curl -X "GET" -s "https://$AWX_ROUTE/api/v2/projects/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure|jq ".results[].name"| sed 's/^/         /'
echo "    "
echo "    -----------------------------------------------------------------------------------------------------------------------------------------------"
echo "    📥 AWX Inventories"
curl -X "GET" -s "https://$AWX_ROUTE/api/v2/inventories/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure|jq ".results[].name"| sed 's/^/         /'
echo "    "
echo "    -----------------------------------------------------------------------------------------------------------------------------------------------"
echo "    📥 AWX Execution Environments"
curl -X "GET" -s "https://$AWX_ROUTE/api/v2/execution_environments/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure|jq ".results[].name"| sed 's/^/         /'
echo "    "
echo "    -----------------------------------------------------------------------------------------------------------------------------------------------"
echo "    📥 AWX Job Templates"
curl -X "GET" -s "https://$AWX_ROUTE/api/v2/job_templates/" -u "$ADMIN_USER:$ADMIN_PASSWORD" --insecure|jq ".results[].name"| sed 's/^/         /'
echo "    "
echo "    "
echo "    "


echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo " 🚀 AWX Access"
echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo " -----------------------------------------------------------------------------------------------------------------------------------------------"
echo "    "
echo "     📥 AWX :"
echo ""
echo "         🌏 URL:      https://$AWX_ROUTE"
echo "         🧑 User:     admin"
echo "         🔐 Password: $(oc -n awx get secret awx-admin-password -o jsonpath='{.data.password}' | base64 --decode && echo)"
echo "    "
echo "    "


echo "*****************************************************************************************************************************"
echo " ✅ DONE"
echo "*****************************************************************************************************************************"

while true
do
    sleep 60000                           
done

